frontend/src/Frontend/UI/Transfer.hs

```diff
@@ -194,8 +199,36 @@ toFormWidget model cfg = mdo
   (clk,(_,k)) <- controlledAccordionItem keysetOpen mempty (accordionHeaderBtn "Owner Keyset") $ do
     keysetFormWidget $ (snd <$> cfg)
       & setValue %~ modSetValue (Just (fmap userFromPactKeyset . _txBuilder_keyset <$> pastedBuilder))
+      & setValue %~ modSetValue (Just $ Nothing <$ eClear)
+
+  let
+    eKSAndChainAddr = attach (current $ value tca) $ updated k
+    eNewPrincipal = fforMaybe eKSAndChainAddr $ \(mCa, mKS) ->
+      case (mCa, mKS) of
+        (Nothing, Just ks) -> Just $ ChainAccount (ChainId "0") $ toPrincipalAccName ks
+        (Just (ChainAccount cid (AccountName an)), Just ks) ->
+          if isPrincipalAccName an
+            then Just $ ChainAccount cid $ toPrincipalAccName  ks
+            else Nothing
+        otherwise -> Nothing
 
   return (tca,k)
+  where
+    isPrincipalAccName = (==) ':' . flip T.index 1
+    toPrincipalAccName = AccountName . createKSPrincipal . userToPactKeyset
+
+createKSPrincipal :: KeySet -> Text
```

Comment:
It would be nice if Pact supplied these (@jwiegley @jmcardon) but this is fine for now.